src/app/frontend/logs/component.js

```diff
@@ -282,14 +282,26 @@ export class LogsController {
   }
 
   /**
-   * Return the link to download the log file
+   * Initiate logs download.
    * @export
-   * @return {string}
    */
-  getDownloadLink() {
+  downloadLog() {
     let namespace = this.stateParams_.objectNamespace;
-    return `api/v1/log/file/${namespace}/${this.pod}/${this.container}?previous=${
-        this.logsService.getPrevious()}`;
+    this.logsService.downloadLogFile(namespace, this.pod, this.container).then((data) => {
+      let url = URL.createObjectURL(data.response);
+      let a = document.createElement('a');
+
+      a.href = url;
+      a.download = this.logsService.getLogFileName(this.pod, this.container);
+      a.target = '_blank';
+
+      // Link has to be added to body in order to fix download in Firefox.
+      if (document.body != null) {
+        document.body.appendChild(a);
```

Comment:
I assume with every click a new `<a> `will be added to the dom. I assume this is ok.

Reply:
I'll optimize it and remove in finally block.